Partner Manager - Reorg of architecture and trading partner topics (#139)
mulesoft/docs-partner-manager: modules/ROOT/pages/edifact-receive-read-settings.adoc

[[ack-settings]]
== Acknowledgment Settings

Optionally configure acknowledgment settings for each Inbound EDIFACT message flow. To configure acknowledgement settings:

*Vijay, please make sure these descriptions are correct.*

* Enable *Send CONTRL technical acknowledgement for each transmission from <partner-name>*.
+
This acknowledgement indicates whether Partner Manager received and accepted the message.

> +

Reviewer: @dhirsch1  - For EDIFACT, there is just one CONTRL acknowledgement. the configuration "CONTRL technical acknowledgement" is being removed from the UI with this release, and we should remove these lines from the Doc.

* Enable the *Send CONTRL functional acknowledgement when messages fail to route to a message flow* setting.
+

> When there is a routing failure, this acknowledgement indicates whether Partner Manager validated the message.

Reviewer: Please rephrase as below:
When there is a routing failure, this acknowledgement indicates that the message is being rejected.

To configure acknowledgements:

. Enable the *Send CONTRL functional acknowledgement when messages fail to route to a message flow* setting.

> . Select an endpoint from which the functional acknowledgement message will be sent to the trading partner.

Reviewer 2: ```suggestion
. Select an endpoint from which the functional acknowledgment message is sent to the trading partner.
```

The *Send CONTRL functional acknowledgement when messages fail to route to a message flow* setting provides a functional acknowledgment (the receipt and syntactical acceptability of a message) with the exchange.

> Functional acknowledgment settings are configured for each message flow and apply only when there is a failure to route the message to one of the message flows.

Reviewer 2: active? Might be too awk...:
```suggestion
Configure for each message flow functional acknowledgment settings that apply if message routing to one of the message flows fails.
```

== Validation settings

> Select one or more of the following options to ensure that all EDIFACT messages received from the trading partner meet validation rules. If a message does not meet an enabled rule, Partner Manager rejects it.

Reviewer 2: I wonder if the table should follow the first sentence here, followed by the the second and the text in Line 35?

Select one of the following...:
[table]
If a message does not match.... Partner Manager logs....

Parter Manager logs all validation errors and reports them in the CONTRL acknowledgments.

[%header%autowidth.spread]
|===
|Setting |Description |Default

|Enforce length limits
a|
* If selected, Partner Manager rejects a message with values that are too short or long.
* If unselected, Partner Manager accepts the message.
|Enabled
|Enforce value repeats
a|
* If selected, Partner Manager rejects a message with values that are repeated too many or too few times.
* If unselected, Partner Manager accepts the message.
|Enabled
|Enforce valid characters

> a| * If selected, Partner Manager rejects a message with invalid characters.

Reviewer 2: ```suggestion
a| * If selected, Partner Manager rejects a message that contains invalid characters.
```

> * If unselected, the characters are either passed through or substituted and Partner Manager accepts the message.

Reviewer 2: ```suggestion
* If unselected, the invalid characters are either ignored or replaced, and Partner Manager accepts the message.
```

|Setting |Description |Default

|Requires unique interchange control number
a| * If selected, Partner Manager records the interchange numbers previously processed and rejects duplicate interchange numbers from the same partner, as determined by the interchange sender and receiver identification.
* If unselected, Partner Manager enables processing of the received interchange to continue.
|Disabled

|Requires unique message control number
a| If selected, Partner Manager enforces globally unique message control numbers (UNH01) for received transaction sets.

> This configuration requires message control numbers to be unique across all EDIFACT messages received from the same partner and application, as determined by the interchange sender and receiver identification, and application codes.

Reviewer 2: ```suggestion
This configuration requires message control numbers to be unique across all EDIFACT messages received from the same partner and application, as determined by the interchange sender and receiver identification and application codes.
```
